ambari-server/src/main/java/org/apache/ambari/server/actionmanager/ActionDBAccessorImpl.java

```diff
@@ -154,6 +156,8 @@
   private Cache<Long, HostRoleCommand> hostRoleCommandCache;
   private long cacheLimit; //may be exceeded to store tasks from one request
 
+  private AtomicBoolean jpaInitialized = new AtomicBoolean(false);
```

Comment:
Can you explain the need for this? There are really only a few odd  cases where this would be needed.

Reply:
The reason I needed go add this is because from now on (if this change gets committed) `ActionDBAccessor` is injected in `TaskUpdateListener' and when the Guice injector is being created in `AmbariServer` the `init()` method has been called prematurely: before starting and initializing JPA context by invoking `injector.getInstance(GuiceJpaInitializer.class)`.  To avoid this we have to wait until this is happening and a `JpaInitializedEvent` is triggered.

I'm open to any recommendation on how to resolve this in another way.

Reply:
Can you instead use a `Provider<>` to prevent this race condition?